### ISSUE TITLE
Samsung TV Model UE40J5200AU video error

### ISSUE THREAD

**Issue by vprocopan** [benzea/gnome-screencast#126]
Hello, please advise me on the following issue. I have previously tried MiraCast protocol in both windows 10 and android and it worked, after running into a few issues and successfully resolving them, I am seeing this error on my tv.
![IMG_20200529_084239](https://user-images.githubusercontent.com/6999055/83226254-3d0c5a80-a18a-11ea-8c47-f75c943239cd.jpg)
Here is the debug log
![IMG_20200529_084254](https://user-images.githubusercontent.com/6999055/83226259-3e3d8780-a18a-11ea-877a-67659125c949.jpg)
Please suggest a remedy, thanks.

**Comment by benzea** [comment]
Hehe, I think I know what the problem is.

I decided to hardcode a FullHD resolution. So gnome-network-manager correctly detects the available resolutions of the monitor, but then just uses FullHD in all cases. And FullHD is not available in your case. I guess in your case it should be picking `1280x768`.

**Comment by vprocopan** [comment]
Hello! Thanks for looking into the issue! So the TV is FullHD compatible which is 1920x1080, and the 2 monitor setup in Ubuntu/Gnome is 1366x768 Built-in Display, and also 1280x1024 VGA Additional Screen. 

Here are the supported resolution it writes:
** (gnome-network-displays:8251): DEBUG: 06:38:14.805: Supported resolutions:
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 800x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 854x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 864x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 640x360 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 960x540 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 848x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 640x480 60p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 720x480 60p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 720x480 60i
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 720x576 50p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 720x576 50i
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 1280x720 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 800x600 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * 1024x768 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.805: Add codec to params:
** (gnome-network-displays:8251): DEBUG: 06:38:14.805: WfdVideoCodec:
** (gnome-network-displays:8251): DEBUG: 06:38:14.805:  * profile: 2
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * level: 4
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * native: 640x480 60p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806: Supported resolutions:
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 800x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 854x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 864x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 640x360 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 960x540 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 848x480 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 640x480 60p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 720x480 60p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 720x480 60i
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 720x576 50p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 720x576 50i
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 1280x720 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 800x600 30p
** (gnome-network-displays:8251): DEBUG: 06:38:14.806:  * 1024x768 30p


Why doesn't it say it support FullHD if the hardware does?
I've also checked which resolution, for example, some app of Android MiraCast uses, and it is 1280x720 (approximately, the one you mentioned)
![IMG_20200530_064256](https://user-images.githubusercontent.com/6999055/83318828-08a7a580-a241-11ea-8775-7c43448e6f5e.jpg)
So in conclusion, what can be done to let it switch the correct resolution? Or auto-assign it for instance? Thanks!

**Comment by benzea** [comment]
Right, if it is really `1366x768` that might explain (FullHD is 1920x1080).

So, FullHD will not be possible. But one can fix the issue that gnome-network-displays does not select a proper resolution. I need to think about how to do the resolution selection implementation then though, and that requires some more considerations (local resolution and EDID information of attached monitor).

EDIT: My 1280x726 was just a typo, I meant to write 1280x720.

**Comment by benzea** [comment]
Said differently, it is easy to patch which resolution is selected, but I really need to think about selecting a good resolution automatically.

**Comment by vprocopan** [comment]
Hello @benzea No rush, but have you looked upon the issue? Else could you specify the line where you hardcoded the resolution so I can adjust! Thanks!

**Comment by benzea** [comment]
It is in `wfd_client_select_codec_and_resolution` in `src/wfd/wfd-client.c`. There is also a version of the code that is commented out using `#if 0` (i.e. changing both `#if 0` to `#if 1` should enable a dumb selection algorithm, I believe).

**Comment by vprocopan** [comment]
Hello @benzea ! So i've changed the hardcoded resolution to my TV's supported 720p but i still got some errors
![image](https://user-images.githubusercontent.com/6999055/84235688-f3fbc500-aafe-11ea-9993-9377dea402c9.png)
Here is the log from the debug, something with openGL i guess, and the tv still shows Video data is not supported.
Any ideas? Thanks!
